### PR TITLE
Add Minimal header component

### PR DESCRIPTION
**Pull request by ataker**
## Chromatic

https://minimal-header-dst2030--60f9b557105290003b387cd5.chromatic.com

## Description
Closes [Minimal header - 2030](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2030)

## Testing done
Tested in Storybook in Chrome, Safari, Edge, and FIrefox

## Screenshots
Screenshot 2023-09-14 at 9 45 16 AM
Screenshot 2023-09-14 at 9 45 23 AM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx:20]: I wonder if it would be useful to have `va-crisis-line-modal` as its own standalone component (instead of embeded)? 🤔

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:7]: Are we still trying to just use `px` instead of `rem` because of the v1/v3 root font size clash? If so, then you will probably need to convert all the em/rem in the CSS to px.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:1]: Does this need to be done? I think it's only needed for USWDS/v3 specific components.

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:59]: I'm guessing this component was just a lift and shift from vets website? We can probably clean up things like this though.

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:3]: This is the React import for CONTACTS (being deprecated). You should use this import instead:

```
import { CONTACTS } from '@department-of-veterans-affairs/component-library/contacts';
```

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss:4]: Is this the right place to be adding tokens? Do we have these values somewhere else in a central place we could pull from instead?

**Comment by jamigibbs** [comment]
It might just be me but is the VA icon kind of blurry?

Screenshot 2023-09-14 at 4 27 30 PM

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx
Comment (jamigibbs) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx:20]: After I commented there I remember that we do have the crisis modal already a variation of va-modal. Is it possible to just use this?

https://minimal-header-dst2030--60f9b557105290003b387cd5.chromatic.com/?path=/docs/components-va-modal--crisis-line-modal

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (micahchiang) [inline packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:33]: I think most of our component styles use numerical values for `font-weight`. It's been a minute since I've seen `lighter` used, so I had to read the [docs](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight) on it. Interesting that its computed value is derived from whatever font-weight is inherited. 

Are we able to use numerical values here? Or, if `lighter` is needed in order to get parity with the mockups, can you leave a comment for why its used on what its computed value ends up being?

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss
Comment (micahchiang) [inline packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss:22]: I think we need `cursor: pointer` somewhere in here. Currently the cursor remains a cursor when hovering over the button

**Review by micahchiang — COMMENTED** [review]
I don't have access to the sketch file that contains the mocks for this. Do we have mockups that handle long header values that get passed in? Currently on mobile, long strings make the header take up a lot of space:

Screenshot from 2023-09-14 15-17-29

Again, I don't have access to the sketch file for this, so not sure if this is expected or not.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx
Comment (micahchiang) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx:19]: Specifically the `Here's how you know span`

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss
Comment (jamigibbs) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss:4]: This might be irrelevant if we are able to just use the existing crisis modal variation from `va-modal`: https://design.va.gov/storybook/?path=/docs/components-va-modal--crisis-line-modal

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (jamigibbs) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:1]: https://design.va.gov/storybook/?path=/docs/components-va-modal--crisis-line-modal

Comment (micahchiang) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:1]: Yeah saw that. We have [this ticket](https://app.zenhub.com/workspaces/platform-design-system-5f8de67192551b0012ebb802/issues/gh/department-of-veterans-affairs/vets-design-system-documentation/2016) to address some crisis line inconsistencies in storybook. It seems like `.va-crisis-line` and `.va-crisis-line-button` renders a different version than what we're using in here.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx:20]: We have [a ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2016) to update that modal to match the crisis line modal in production. So my hope in making this would be to help accomplish that update. Maybe we update that ticket into taking this and making it a standalone component?

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:59]: Removed

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:1]: Right, this was a remnant for earlier, removed.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:7]: Converted to px

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx:3]: Thanks! I was hoping there was somewhere better to grab this from. Updated import.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.tsx:19]: I opened [a new ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2097) to address this.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:3]: Yeah, this wouldn't collide, but I agree it's a bad name. Updated.

File: packages/web-components/src/components/va-minimal-header/va-minimal-header.scss
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-minimal-header.scss:33]: Updated to numerical.

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.scss:22]: Added

**Comment by ataker** [comment]
Updated logo to an SVG I found from va.gov

Line height reduced for the header

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (ataker) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx]: Hmm, that import does not work in the component library for me...

**Comment by Andrew565** [comment]
This PR is ready for re-review, and now also includes fixes for #[2016](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2016).

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:16]: Can we have some tests written for this component?

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:2]: Is this the right place for color tokens?

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:4]: This token also seems like it should not be part of any individual component.

**Comment by jamigibbs** [comment]
Do we happen to have a sketch or design file for the crisis line modal? I'm not sure what we should be checking it against: 

Screenshot 2023-09-25 at 3 01 13 PM

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (jamigibbs) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:3]: Alternately, the direct path should be something like (it's in the `packages/web-components/src/` folder)

```
import { CONTACTS } from '../../contacts.ts';
```

**Comment by Andrew565** [comment]
@jamigibbs Right now, I think the only thing we have to compare it to is the existing crisis line modal at va.gov.

**Comment by jamigibbs** [comment]
> @jamigibbs Right now, I think the only thing we have to compare it to is the existing crisis line modal at va.gov.

No problem. So if that's the case, this is what va.gov is showing. The modal looks wider than what I see in the Chromatic link. Should the va-modal have the `large` prop set?

Screenshot 2023-09-26 at 1 54 34 PM

**Comment by jamigibbs** [comment]
There is also this auto-focus on the first phone number that I don't see in our component. I'm assuming we want to keep that behavior?

Screenshot 2023-09-26 at 1 56 13 PM

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (Andrew565) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:16]: Good idea, on it!

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (Andrew565) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:109]: Also a good idea!

File: packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx
Comment (Andrew565) [reply packages/web-components/src/components/va-minimal-header/va-crisis-line-modal.tsx]: I changed this to pull from /web-components/index.ts which exports CONTACTS and it appears to work.

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss
Comment (Andrew565) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:4]: I thought about it and couldn't find a better solution for tokens than what Alex did here. These values just don't exist anywhere else that we can use right now, until the remainder of the USWDS tokens are brought into the variables file.

Comment (jamigibbs) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:4]: Right now we have a tokens file in the [web-components directory](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/global/_variables.scss), in the [CSS Library](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/css-library/dist/tokens/scss/variables.scss), and in [Formation](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/9578cd69766b7b75dce415188b73f90c79eb85ea/packages/formation/sass/base/_b-variables.scss#L132). Now we are adding random tokens here in a specific component.

It just feels like there are tokens everywhere and adding more here might make dealing with them later more challenging than it already will be.

Can we consider adding the colors we need to the CSS-Library, regenerate the file, and then updating the tokens file in the web-components directory with the new compiled file?

File: packages/web-components/src/components/va-crisis-line-modal/test/va-crisis-line-modal.e2e.ts
Comment (jamigibbs) [inline packages/web-components/src/components/va-crisis-line-modal/test/va-crisis-line-modal.e2e.ts:7]: Is this test checking anything?

File: packages/web-components/stencil.config.ts
Comment (jamigibbs) [inline packages/web-components/stencil.config.ts:60]: Do you mind mentioning why we need this or what it does?

File: packages/web-components/src/contacts.ts
Comment (jamigibbs) [inline packages/web-components/src/contacts.ts:96]: Was this giving some Typescript errors before that this addition is resolving?

**Comment by jamigibbs** [comment]
There are just a couple of visual things I noticed

There is a bit larger spacing between the header and content than what is on the VA.gov version.

And there is an icon next to the header there that I don't see here:

**VA.gov Example**

270765968-32aab485-d59a-49f0-8216-c62a1e84ade0


**Chromatic**

Screenshot 2023-09-29 at 2 52 39 PM

**Comment by jamigibbs** [comment]
The Storybook example does not display the code markup like other component stories do:

Screenshot 2023-09-29 at 2 56 50 PM

File: packages/web-components/src/contacts.ts
Comment (Andrew565) [reply packages/web-components/src/contacts.ts:96]: Yes, previously it resolved as type `Readonly{ {} }` which meant that it didn't see any of the Object Keys for the ContactsMap.

**Comment by Andrew565** [comment]
@jamigibbs I fixed the CLM's modal padding so there's no more gap, and also fixed it so that the code snippet shows correctly. I'm unable to add the flag icon to the modal title at this time as that prop only accepts strings and not JSX.

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (micahchiang) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:88]: The number portion of this link should be bold I think: 

image

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (micahchiang) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:111]: Don't think we need the `tty` prop here, even though it's technically a tty number. This is what's on production:

image

**Review by micahchiang — COMMENTED** [review]
Left some comments on link styling for the crisis line modal. Also, I noticed a couple things while tabbing through the modal: 
1. The focus color changes between green and yellow, not sure if this is an issue with my browser or something else. 
2. The tab order seems to be inconsistent with what's on production. In production, tab order goes to the modal close button after the last link in the list. In Chromatic, tab order exits the modal completely and I had to shift + tab back all the way through the modal links in order to get to the modal close button. 

Here's a gif displaying both things:
Screencast from 10-03-2023 10 34 56 AM

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss
Comment (jamigibbs) [inline packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:12]: Something I just noticed is that when the crisis line button is used in isolation, it displays with padding on the left and right I think because of this max-width.

Screenshot 2023-10-03 at 3 34 56 PM

As an isolated component (outside of it being used in the header), do we want the component internals to be managing positioning like that? Or should that be left to the parent to control as needed (the minimal header in this case)?

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (Andrew565) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:88]: The number portion of that link is already bold.

Comment (micahchiang) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:88]: Sorry, I realize my screenshot probably wasn't very helpful haha. 

This is what I see on production: 
image

And this is what I see on Chromatic:
Screenshot from 2023-10-04 09-13-19

Production also includes `Text` as part of the link.

Comment (jamigibbs) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:88]: This might be a situation where we want to align the visual display of numbers based on what `va-telephone` is doing rather than try and duplicate what the existing VA.gov modal is doing.

So `<va-telephone sms` and `<va-telephone tty` are the desired visual displays even though they are different from what VA.gov has hard coded? 

With that said (and looking at [va-telelphone examples](https://design.va.gov/storybook/?path=/docs/components-va-telephone--default)), it seems like none of the numbers should be bold. We might want a designer to confirm though.

Comment (micahchiang) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx:88]: Yeah that's a good point, we can ask in standup about this.

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx
Comment (humancompanion-usds) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.tsx]: That seems to me to be a bug in what is in production. We added the TTY prompt so that folks would be warned before calling a TTY number.

File: packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss
Comment (jamigibbs) [reply packages/web-components/src/components/va-crisis-line-modal/va-crisis-line-modal.scss:15]: @Andrew565 Do you have any thoughts about this spacing issue?

File: packages/web-components/src/components/va-crisis-line-modal/test/va-crisis-line-modal.e2e.ts
Comment (ataker) [reply packages/web-components/src/components/va-crisis-line-modal/test/va-crisis-line-modal.e2e.ts:7]: Test updated

File: packages/web-components/stencil.config.ts
Comment (ataker) [reply packages/web-components/stencil.config.ts:60]: This is so the jest spec test can import the svg without error. We haven't needed this elsewhere because we've been using e2e tests for almost all tests. The spec tests run faster though. [Docs](https://stenciljs.com/docs/unit-testing)

**Comment by ataker** [comment]
Addressed @jamigibbs's feedback

**Review by jamigibbs — APPROVED** [review]
Thank you @ataker! Let's just make sure that we have a designer approval on this too before it gets merged cc @babsdenney @danbrady @LWWright7

**Comment by babsdenney** [comment]
| Image | Storybook | Sketch |
| :--- | :---- | :---- |
| image    |   Button background color (text side) - #b51d09 <br /> <br />  Font style - Arial  <br /> Height of the button - 34px   | Button background color (text side) - #981B1E <br /> Font style - There doesn't seem to be a font style being applied <br /> On VA.gov there the text style in Source sans pro <br /> Height of the button - 32px  |
| image | The medium view has an issue with the .gov banner link |  |
| image | .gov banner height - 30px |  .gov banner height - 22px |
|image | The crisis line modal icon is present | imageThe crisis line icon is removed on small screens |
|| Expand link for the .gov banner is present | The expand link for the .gov banner merges with the text to become a link (see above) <br /> NOTE: This diverges from the USWDS .gov banner that has the expand link present in small screens |
|image | Storybook showes a 16px space between the logo, vertical line, and title | The sketch document varies is space from 7px - 10px but would think instead that 8px should be used for horizontal spacing between the logo, vertical line, and title |
| image | Storybook is showing more vertical padding on the blue header 20px padding on top and bottom | Sketch is showing padding of 10px on top and 11px bottom but maybe we can adjust that to 10px on top and bottom instead | 
| image | The divider is 22px in Storybook | The divider is 38px in Sketch |
| image | The background color for the blue header is #182c54 | The background color for the blue header in Sketch is #112E51 |

**Comment by humancompanion-usds** [comment]
> Storybook showes a 16px space between the logo, vertical line, and title
> The sketch document varies is space from 7px - 10px but would think instead that 8px should be used for horizontal spacing between the logo, vertical line, and title

I just checked the Sketch file and it looks like what I intended to me: 8px at mobile, 16px at desktop between the vertical line and the elements near it. 

I will bring the minimal header from the reference Sketch file into the Component library.

**Comment by humancompanion-usds** [comment]
The divider height was meant to stretch depending on the height of the title. For example:

Screenshot 2023-10-17 at 11 28 28 AM

**Review by micahchiang — APPROVED** [review]
Checked everything in storybook, and it looks good to me :+1: 

There's an issue with the official gov banner at the medium break point in story book, but we have an existing ticket to address that here - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2097

**Comment by babsdenney** [comment]
@Andrew565 I took a look at the new header. Everything looks great except the crisis line button. The color is a bit off and there's also some hover over states that could be adjusted. The Crisis line button is something that hasn't changed, could you match all the specifications from the production crisis line button. The simple header crisis line button can be identical to the VA.gov current header. 

https://www.va.gov/

**Comment by babsdenney** [comment]
@Andrew565 This looks so great! Thank you for making the changes! The only little nit-picky thing I see is the hover-over effect on the icon. It is a tad brighter than the one in the main VA header. Looking at the color they chose, it is not a color in the Design System. So that might be something to figure out in the future. 

They are using - #b51d09

Could we change that color to what is in VA.gov header in prod?

**Comment by humancompanion-usds** [comment]
>  Looking at the color they chose, it is not a color in the Design System.

I see we already made this change but, this is a bug. It is never, ever correct to use a color not in the Design System. Period. "They" here is VA, which is us. It's on us to fix it and not propagate it. I understand we're trying to recreate the current version but in future if the current version is broken or wrong we should address those things. Let's file an issue on this to correct it in future.

**Comment by babsdenney** [comment]
@humancompanion-usds Thanks for calling out! I definitely agree that should have been a flag to figure out how we could align with colors in the Design System. I made a design ticket for that work. 

https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2221

**Review by babsdenney — APPROVED** [review]
Design approved!

**Pull request merged**
